packages/ember-data/tests/integration/store-test.js

```diff
@@ -254,22 +253,58 @@ test("Using store#findRecord on existing record reloads it", function() {
         }
       }
     });
-    car = store.peekRecord('car', 1);
   });
 
   ajaxResponse({
     cars: [{
       id: 1,
-      make: 'BMCW',
-      model: 'Mini'
+      make: 'BMC',
+      model: 'Princess'
+    }]
+  });
+
+  run(function() {
+    store.findRecord('car', 1).then(function(car) {
+      equal(car.get('model'), 'Mini', 'cached car record is returned');
+    });
+  });
+
+  run(function() {
+    let car = store.peekRecord('car', 1);
+    equal(car.get('model'), 'Princess', 'car record was reloaded');
+  });
+});
+
+test("store#findRecord { reload: true } ignores cached record and reloads record from server", function() {
```

Comment:
@HeroicEric in this test could you also assert that `shouldReloadRecord` is never called when `{ reload: true }` is passed into findRecord?